### PR TITLE
[GEOS-11080] Remove ASCII grid output format from WCS

### PR DESCRIPTION
**Pull request by aaime**
[GEOS-11080](https://osgeo-org.atlassian.net/browse/GEOS-11080)

See ticket and discussion on gs-devel list.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by bradh** [comment]
```
 17:26:18,679 [INFO] PMD Failure: org.geoserver.wcs.GetCoverageTest:234 Rule:UnusedLocalVariable Priority:3 Avoid unused local variables such as 'content'..
17:26:18,679 [INFO] PMD Failure: org.geoserver.wcs.GetCoverageTest:234 Rule:UnusedLocalVariable Priority:3 Avoid unused local variables such as 'content'..
```

File: src/wcs1_0/src/test/java/org/geoserver/wcs/GetCoverageTest.java [hunk 1]
```diff
@@ -222,81 +222,25 @@ public void testWorkspaceQualified() throws Exception {
...
 
     @Test
-    public void testGEOS6540_1() throws Exception {
+    public void testGEOS6540() throws Exception {
         String queryString =
                 "wcs?sourcecoverage="
                         + getLayerId(MOSAIC)
                         + "&request=getcoverage"
-                        + "&service=wcs&version=1.0.0&format=ArcGrid&crs=EPSG:4326"
+                        + "&service=wcs&version=1.0.0&format=image/geotiff&crs=EPSG:4326"
```
Comment (bradh) [inline src/wcs1_0/src/test/java/org/geoserver/wcs/GetCoverageTest.java:230]: These tests were checking for the `format` parameter being case-insensitive. 

See https://osgeo-org.atlassian.net/browse/GEOS-6540

File: src/wcs1_0/src/test/java/org/geoserver/wcs/GetCoverageTest.java [hunk 2]
```diff
@@ -222,81 +222,25 @@ public void testWorkspaceQualified() throws Exception {
...
                         + "&bbox=0,0,1,1&width=50&height=60";
 
         MockHttpServletResponse response = getAsServletResponse(queryString);
-        assertEquals("text/plain", response.getContentType());
         String content = response.getContentAsString();
```
Comment (bradh) [inline src/wcs1_0/src/test/java/org/geoserver/wcs/GetCoverageTest.java:234]:
```suggestion
```

(since this is the part failing PMD)

**Comment by sikeoka** [comment]
I've seen `GML grid` mentioned a few times in relation to this but I can't find anything about it in the documentation under WCS, extensions or community modules.  Could you possibly update the documentation to make it easier for users to find and use this `GML grid` format?  Also, references to ArcGrid should be removed from https://docs.geoserver.org/latest/en/user/services/wcs/outputformats.html

File: src/wcs1_0/src/test/java/org/geoserver/wcs/GetCoverageTest.java (hunk 1 above)
Comment (aaime) [reply src/wcs1_0/src/test/java/org/geoserver/wcs/GetCoverageTest.java:230]: I'm going to rename the test methods to make that clear.

**Comment by aaime** [comment]
@sikeoka documentation has been updated. Here is also an example of a 3 band output from one of GeoServer built-in demo data sets here:

https://gs-main.geosolutionsgroup.com/geoserver/nurc/wcs?service=WCS&version=2.0.0&request=GetCoverage&coverageId=nurc__Pk50095&format=application/gml%2Bxml

**Comment by sikeoka** [comment]
I don't see any documentation updates in this PR but I played around with the GML format and it is pretty interesting. The rest of the PR looks fine to me.

**Comment by aaime** [comment]
Ah, forgot to push, now it's in.

**Pull request merged**